### PR TITLE
Start wiring up the scenario sidebar

### LINKED ISSUE #131: Wire the Scenario Sidebar to Read From and Write To Firestore
Screen Shot 2020-04-17 at 3 07 19 PM

Populate the Scenario Sidebar with information from the Cloud Firestore.  You can get information about the baseline scenario from Firestore by using the following code:

```javascript
const baselineScenarioRef = await getBaselineScenarioRef();
const result = await baselineScenarioRef.get();
const scenario = result.data();
```

Note `getBaselineScenarioRef` can be imported from our [db code](https://github.com/Recidiviz/covid19-dashboard/blob/master/src/database/index.tsx#L135).

For this mockup you'll need the `name`, `updatedAt`, `dailyReports`, and `dataSharing` attributes found on the `scenario` object.

As a final step, when the user toggles the `dailyReports`, and `dataSharing` toggles we should save these updates to Firestore by calling the `saveScenario` method that can also be imported from our [db code](https://github.com/Recidiviz/covid19-dashboard/blob/master/src/database/index.tsx#L210).

```javavscript
  saveScenario({
    dataSharing: true,
  });
```

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

> Pass in the baseline scenario to the Scenario Sidebar component. I'm opening this as a WIP and having turned off the linter requirements so that it can be picked up by someone else later if needed. 

TODO: 

- [ ] Update the typescript types for the component props
- [x] Add `dailyReports` and `dataSharing` to the `scenario` model 
- [x] Add a data formatter for the `updatedAt` data 
- [x] Save the daily reports & data sharing booleans to the scenario

Question - What data should go in the description below the header?

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #131

### Development

These boxes should be checked by the submitter prior to merging:

- [ ] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/database/index.tsx [hunk 1]
```diff
@@ -192,6 +192,8 @@ export const createBaselineScenario = async () => {
     baselineScenarioRef = await db.collection(scenariosCollectionId).add({
       name: "Baseline Scenario",
       baseline: true,
+      dataSharing: false,
+      dailyReports: false,
```
Comment (sprad) [inline src/database/index.tsx:196]: Thanks for adding these.  I've manually updated the existing scenarios in Firebase to set these values

File: src/page-multi-facility/ScenarioSidebar.tsx [hunk 2]
```diff
@@ -1,11 +1,18 @@
 import PromoBoxWithButton from "../design-system/PromoBoxWithButton";
 import ToggleRow from "./ToggleRow";
+import { saveScenario } from "../database";
+
+interface ScenarioSidebar = {
+  scenario: Object;
```
Comment (sprad) [inline src/page-multi-facility/ScenarioSidebar.tsx:6]: Feel free to add a proper `Scenario` type in this file:  https://github.com/Recidiviz/covid19-dashboard/blob/master/src/page-multi-facility/types.tsx

Comment (sprad) [reply src/page-multi-facility/ScenarioSidebar.tsx:6]: Will look something like this:

```javascript
export type Scenario = {
  name: string;
  baseline: boolean;
  dataSharing?: boolean;
  dailyReports?: boolean;
  roles: Map<String, String>;
  facilities?: Facilities;
};
```

Comment (sprad) [reply src/page-multi-facility/ScenarioSidebar.tsx:6]: Will also include  `createdAt` and  `updatedAt` as required fields.  They're stored as `firebase.firestore.FieldValue.serverTimestamp();`, but I'm not sure what that translates to in the TypeScript/JavaScript Datetime world

File: src/page-multi-facility/MultiFacilityImpactDashboard.tsx [hunk 3]
```diff
@@ -20,15 +20,32 @@ const MultiFacilityImpactDashboardContainer = styled.main.attrs({
...
     data: [] as Facilities,
     loading: true,
   });
 
+  const [scenario, setScenario] = useState({
+    data: {},
+    loading: true
+  })
+
   useEffect(() => {
```
Comment (sprad) [inline src/page-multi-facility/MultiFacilityImpactDashboard.tsx:40]: I don't know enough about React, but is each piece of managed state supposed to get its own `useEffect` or is it common to nest multiple calls under a single `useEffect` like we're doing here?

File: src/page-multi-facility/MultiFacilityPage.tsx [hunk 4]
```diff
@@ -10,19 +10,25 @@ import MultiFacilityImpactDashboard from "./MultiFacilityImpactDashboard";
 
 const MultiFacilityPageDiv = styled.div``;
 
+type ScenarioType = {
+  data: object | null;
+  lodaing: boolean;
```
Comment (sprad) [inline src/page-multi-facility/MultiFacilityPage.tsx:15]: Nit:  "lodaing" -> "loading"

File: src/page-multi-facility/ScenarioSidebar.tsx (hunk 2 above)
Comment (daschi) [reply src/page-multi-facility/ScenarioSidebar.tsx:6]: Okay I can add these!

File: src/page-multi-facility/MultiFacilityImpactDashboard.tsx (hunk 3 above)
Comment (jessex) [reply src/page-multi-facility/MultiFacilityImpactDashboard.tsx:40]: I think the way to think of it is as discrete operations. If you have two things you want to happen independently of each other, they should be their own hooks. If you have two things that you want to happen in tandem or in a particular order, they may live in the same hook.

File: src/page-multi-facility/ScenarioSidebar.tsx [hunk 5]
```diff
@@ -15,15 +22,17 @@ const ScenarioSidebar: React.FC = () => {
           </p>
         </div>
         <div>
-          <p className="text-xs text-gray-500">Last Update: March 25, 2020</p>
+          <p className="text-xs text-gray-500">Last Update:</p>
```
Comment (sprad) [inline src/page-multi-facility/ScenarioSidebar.tsx:25]: I'm not 100% sure how Firestore returns date times so you'll likely need to do some formatting here.  The `DateFormats` file may be of service:  https://github.com/Recidiviz/covid19-dashboard/blob/94e5cafa080eeebc63aadd032f1a94643f8a403a/src/design-system/DateFormats.tsx

File: src/page-multi-facility/MultiFacilityImpactDashboard.tsx (hunk 3 above)
Comment (jessex) [reply src/page-multi-facility/MultiFacilityImpactDashboard.tsx:40]: In this case, I think maybe the thinking was that loading one or the other but not both might be a poor experience so we should place it within a single hook?

File: src/page-multi-facility/ScenarioSidebar.tsx [hunk 6]
```diff
@@ -15,15 +22,17 @@ const ScenarioSidebar: React.FC = () => {
...
         </div>
         <div className="mt-5 mb-5 border-b border-gray-300" />
         <div>
           <ToggleRow
+            data={dailyReports}
             label="Daily Reports"
             labelHelp="Tooltip help Lorem ipsum dolor sit amet, consectetur adipiscing elit"
```
Comment (sprad) [inline src/page-multi-facility/ScenarioSidebar.tsx:32]: We should get final copy for all of the "lorem ipsum" text in the side bar as well

Comment (jessex) [reply src/page-multi-facility/ScenarioSidebar.tsx:32]: I'm going to take a pass through and update all copy this weekend!

File: src/page-multi-facility/MultiFacilityImpactDashboard.tsx (hunk 3 above)
Comment (daschi) [reply src/page-multi-facility/MultiFacilityImpactDashboard.tsx:43]: Ah okay--I actually just updated it to be 2 hooks before I saw this comment. I think will mean the facilities fetching will not depend on the scenario fetching to complete first, so I think they would happen independently from each other. If we want to ensure both are loaded before loading the content maybe that would be more of a conditional rendering?

File: src/page-multi-facility/ScenarioSidebar.tsx [hunk 7]
```diff
@@ -1,11 +1,21 @@
...
+import { Scenario } from "./types";
+
+interface Props {
+  scenario?: Scenario | null;
+}
+
+const ScenarioSidebar: React.FC<Props> = (props) => {
+  // Typescript error that these properties do not exist on type Scenario
+  // Have I defined the type wrong?
+  const { name, dailyReports, dataSharing } = props.scenario;
```
Comment (daschi) [inline src/page-multi-facility/ScenarioSidebar.tsx:13]: I'm still having issues here with the typescript error saying that `Scenario` type does not have these properties.

File: src/page-multi-facility/ScenarioSidebar.tsx [hunk 8]
```diff
@@ -12,6 +13,15 @@ const ScenarioSidebar: React.FC<Props> = (props) => {
   const { scenario } = props;
   const updatedAtDate = Number(scenario?.updatedAt.toDate());
 
+  const handleToggle = (toggleValue: keyof Scenario) => {
+    const value = scenario?.[toggleValue];
+    // TODO: I think the default value is still set to false for both dataSharing and dailyReports
```
Comment (jessex) [inline src/page-multi-facility/ScenarioSidebar.tsx:18]: Correct: for new scenarios these should default to / start out as false.

**Comment by jessex** [comment]
@daschi, just noticed your question:

>Question - What data should go in the description below the header?

We need to add a description field to the scenario model and make that text editable from the frontend. If you're already in this realm and are comfortable making that addition then great! Otherwise, it can be deferred to another PR.

File: src/database/index.tsx (hunk 1 above)
Comment (sprad) [inline src/database/index.tsx:196]: @daschi - We can add a `description` field and default text here. 
@jessex - Do we have copy we can include as a default description?

Comment (jessex) [reply src/database/index.tsx:196]: Yep! You can use this:
> Welcome to your new scenario. To get started, add in facility data on the right-hand side of the page. Your initial scenario is also your 'Baseline' - meaning this is where you should keep real-world numbers about the current state of your facilities, their cases, and mitigation steps.

I know this was also used in the fresh start modal but I think that may have been a misunderstanding. I'm going to make sure all copy is where it should be later this weekend, so you can use the above for now.

Comment (sprad) [reply src/database/index.tsx:196]: Whoops, accidentally hit Resolve.  Meant to add a comment to let you all know that I added a `description` to the existing Scenarios in firestore with some generic "Default description" copy.

Comment (jessex) [reply src/database/index.tsx:196]: Nice! So I think that means we should be unblocked to add description text reads and updates in the sidebar, @daschi.

File: src/page-multi-facility/types.tsx [hunk 9]
```diff
@@ -6,3 +6,19 @@ export type Facility = {
...
+  name: string;
+  baseline: boolean;
+  dataSharing?: boolean;
+  dailyReports?: boolean;
+  roles: Map<string, string>;
+  createdAt: TimestampType;
+  updatedAt: TimestampType;
+};
+
+type TimestampType = {
```
Comment (sprad) [inline src/page-multi-facility/types.tsx:20]: I'd probably just call this `TimeStamp` unless there is a technical reason to add on the `Type` suffix.

File: src/page-multi-facility/MultiFacilityImpactDashboard.tsx (hunk 3 above)
Comment (jessex) [reply src/page-multi-facility/MultiFacilityImpactDashboard.tsx:51]: I think it's likely a non-issue as currently constructed, but it should become apparent quickly in final testing of so. Let's leave it as is!

File: src/page-multi-facility/ScenarioSidebar.tsx [hunk 10]
```diff
@@ -1,29 +1,60 @@
...
-            This text describes unique qualities for this data model. Taking
-            notes here will be useful when you have multiple data model
-            experiments in your library?
-          </p>
+          <InputTextArea
+            label="Description"
+            value={scenario?.description}
+            placeholder=""
+            onChange={(event) => {
+              // Question - Should we have a Save button to save the description?
```
Comment (jessex) [inline src/page-multi-facility/ScenarioSidebar.tsx:32]: I don't see a save button in the mocks, so I think it should just persist the description change automatically when finished.

**Comment by itsjon** [comment]
@jessex thank you! For now I'm deferring most of the linting issues unless they're build-breaking.

**Comment by daschi** [comment]
Happy to see it was merged, thanks @itsjon and @jessex!

**Pull request merged**